woocommerce-services.php

```diff
@@ -528,6 +528,175 @@ public function init_core_wizard_payments_config() {
 			}
 		}
 
+		/**
+		 * Get WCS PayPal proxy endpoint
+		 */
+		public function paypal_ec_endpoint() {
+			$ppec_settings = get_option( 'woocommerce_ppec_paypal_settings', array() );
+			return trailingslashit( WOOCOMMERCE_CONNECT_SERVER_URL ) . 'paypal/nvp/' . $ppec_settings[ 'environment' ];
+		}
+
+		/**
+		 * Limit supported payment gateway features to payments
+		 */
+		public function paypal_ec_supports( $supported, $feature, $gateway ) {
+			return 'ppec_paypal' === $gateway->id ? 'products' === $feature : $supported;
+		}
+
+		/**
+		 * Once a payment is received, show prompt to connect a PayPal account on certain screens
+		 */
+		public function paypal_ec_prompt_to_connect() {
+			function uses_ppec_payment_method( $order ) {
+				return $order->get_payment_method() === 'ppec_paypal';  // TODO WC <3.0 compatibility
+			}
+			if ( ! count( array_filter( wc_get_orders( array() ), 'uses_ppec_payment_method' ) ) ) {
```

Comment:
This is an expensive check to run on every `admin_notices` action ([#](https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/blob/master/includes/class-wc-gateway-ppec-plugin.php#L175)).

I know this logic is likely to change significantly - since we want to display the banner after an order comes in, perhaps it makes more sense to hook into order-based actions instead of querying all orders.